recipes/lora_finetune_single_device.py

```diff
@@ -405,10 +410,22 @@ def save_checkpoint(self, epoch: int) -> None:
             k: v for k, v in self._model.state_dict().items() if adapter_key_filter(k)
         }
         ckpt_dict.update({utils.ADAPTER_KEY: adapter_state_dict})
+        adapter_config = {
+            "r": self._lora_rank,
+            "lora_alpha": self._lora_alpha,
+            "target_modules": get_lora_module_names(
+                self._lora_attn_modules,
+                self._apply_lora_to_mlp,
+                self._apply_lora_to_output,
+            ),
+            "peft_type": "LORA",
```

Comment:
Nice!

Reply:
Ah good point. I was trying to avoid this initially since it may necessitate some changes to our load_checkpoint [method](https://github.com/pytorch/torchtune/blob/dc2b9911cf5a793a8b4f9cd4df321c5271745b4a/torchtune/utils/_checkpointing/_checkpointer.py#L371), as right now we really only retrieve and remap model weights. If it's more of a nice-to-have, I may punt on it for this particular PR to keep things more isolated to save_checkpoint. Lmk if this makes sense. Also cc @kartikayk if you have any general thoughts on loading state/metadata through `load_checkpointer` and passing through our recipe. I imagine this is something we may want to start supporting more for various integrations anyways.

Reply:
Can you expand a bit more on why we would need ```base_model_name_or_path```? Is this to make sure there are no bugs related to selecting the right base model for further training in HF land? If so, I wonder if this is something which is a "must have" rather than a "good to have"? or let me know if I misunderstand?

If it's a must have, then is this something we can read from one of the json files or do we need to pass this information along through the recipe?

Reply:
We don't strictly need `base_model_name_or_path`, but not having it means that the burden is on the user to figure out which base model this adapter belongs to. Of course, this can be solved with good documentation, but having it automatically in the `adapter_config.json` would be quite convenient.

Other points to consider:
- When shared on HF Hub, this metadata can be used for other things (I'm not an expert on this though)
- If `base_model_name_or_path` is present, users can load the adapter + base model in a single line of code (e.g. `AutoModelForCausalLM.from_pretrained(<path-to-adapter>)`).